### ISSUE TITLE
Optional chaining causes bugs false positive with react/display-name

### ISSUE THREAD

**Issue by Karolusrex** [yannickcr/eslint-plugin-react#2571]
Minimal example:

test.jsx:
```

import PropTypes from 'prop-types';
import React from 'react';

function MyComponent({ myProp }) {
  return myProp?.map(subProp => <p>subProp</p>);
}

MyComponent.propTypes = {
  myProp: PropTypes.arrayOf(PropTypes.string),
};

MyComponent.defaultProps = {
  myProp: null,
};

```

.eslintrc.js:

```
module.exports = {
  parser: 'babel-eslint',
  root: true,
  env: {
    browser: true,
  },
  settings: {
    react: {
      version: 'detect'
    }
  },
  extends: [
    'plugin:react/recommended',
  ],
};
```

`> ./node_modules/.bin/eslint test.jsx`

```
/x/test.jsx
  5:22  error  Component definition is missing display name  react/display-name
```

babel-eslint: 10.0.3
eslint: 6.8.0
eslint-plugin-react: 7.14.3

**Comment by ljharb** [comment]
cc @alexzherdev

We might have a number of things that need updates for optional chaining.

**Comment by jzabala** [comment]
This things to be working with versions:

```
"babel-eslint": "^10.1.0",
"eslint": "^7.4.0",
"eslint-plugin-react": "^7.20.3"
```

![Screen Shot 2020-07-09 at 10 05 09 PM](https://user-images.githubusercontent.com/1315054/87108215-bee2af80-c22f-11ea-98c3-6f553488946c.png)

**Comment by ljharb** [comment]
Closing; happy to reopen if it's not actually fixed.
